arxiv/arxiv.py

```diff
@@ -222,11 +224,13 @@ def download_source(self, dirpath: str = "./", filename: str = "") -> str:
             filename = self._get_default_filename("tar.gz")
         path = os.path.join(dirpath, filename)
         # Bodge: construct the source URL from the PDF URL.
-        source_url = self.pdf_url.replace("/pdf/", "/src/")
+        # FIXME: `self.pdf_url` is possibly `None` and following code raises `TypeError`.
+        source_url = self.pdf_url.replace("/pdf/", "/src/")  # type: ignore[union-attr]
```

Comment:
`arxiv.Result._get_pdf_url([])` returns `None` and it raises `TypeError` in this line.